File: SingleTon/app/src/main/java/com/example/singleton/MainActivity.java
package com.example.singleton;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    EditText editText;
    Button Save;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        editText=findViewById(R.id.editText);
        Save=findViewById(R.id.save);
        Save.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(editText.getText().toString().isEmpty()) {

                    Toast.makeText(MainActivity.this, "Enter Text", Toast.LENGTH_SHORT).show();
                   // editText.setError("enter text");
                }else{
                    String editValue=editText.getText().toString();
                    SingleTon singleTon=com.example.singleton.SingleTon.getInstance();
                    singleTon.setText(editValue);
                    Toast.makeText(MainActivity.this,singleTon.getText(), Toast.LENGTH_SHORT).show();
                }
            }
        });
    }
}

File: DiceRollere/app/src/main/java/com/example/dicerollere/MainActivity.kt
package com.example.dicerollere

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast

class MainActivity : AppCompatActivity() {// Appcompt is subclass of Activity. used to app available to largest number of devices and users possible
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)//R class includes all the apps assets including content of the res
   val rollButton:Button=findViewById(R.id.button)
    rollButton.setOnClickListener{rollDice()}

    }

    private fun rollDice() {
        Toast.makeText(this,"Dice Rolled",Toast.LENGTH_SHORT).show()

        val randomInt=(1..40).random() // method to get a random number between 1 and 6
        val resultText:TextView=findViewById(R.id.txtView)

        resultText.text=randomInt.toString()


    }
}